test/core/application/app_config_test.cpp

```diff
@@ -150,59 +277,108 @@ TEST_F(AppConfigurationTest, LevelDBPathTest) {
   }
   {
-    char const *args[] = {"/path/", "--verbosity", "3"};
-    app_config_->initialize_from_args(
-        AppConfiguration::LoadScheme::kFullSyncing,
-        sizeof(args) / sizeof(args[0]),
-        (char **)args);
+    char const *args[] = {"/path/",
+                          "--verbosity",
+                          "3",
+                          "--genesis",
+                          "genesis_path",
+                          "--leveldb",
+                          "leveldb_path",
+                          "--keystore",
+                          "keystore path"};
+    app_config_->initialize_from_args(AppConfiguration::LoadScheme::kValidating,
+                                      sizeof(args) / sizeof(args[0]),
+                                      (char **)args);
     ASSERT_EQ(app_config_->verbosity(), spdlog::level::level_enum::warn);
   }
   {
-    char const *args[] = {"/path/", "--verbosity", "4"};
-    app_config_->initialize_from_args(
-        AppConfiguration::LoadScheme::kFullSyncing,
-        sizeof(args) / sizeof(args[0]),
-        (char **)args);
+    char const *args[] = {"/path/",
+                          "--verbosity",
+                          "4",
+                          "--genesis",
+                          "genesis_path",
+                          "--leveldb",
+                          "leveldb_path",
+                          "--keystore",
+                          "keystore path"};
+    app_config_->initialize_from_args(AppConfiguration::LoadScheme::kValidating,
+                                      sizeof(args) / sizeof(args[0]),
+                                      (char **)args);
     ASSERT_EQ(app_config_->verbosity(), spdlog::level::level_enum::err);
   }
   {
-    char const *args[] = {"/path/", "--verbosity", "5"};
-    app_config_->initialize_from_args(
-        AppConfiguration::LoadScheme::kFullSyncing,
-        sizeof(args) / sizeof(args[0]),
-        (char **)args);
+    char const *args[] = {"/path/",
+                          "--verbosity",
+                          "5",
+                          "--genesis",
+                          "genesis_path",
+                          "--leveldb",
+                          "leveldb_path",
+                          "--keystore",
+                          "keystore path"};
+    app_config_->initialize_from_args(AppConfiguration::LoadScheme::kValidating,
+                                      sizeof(args) / sizeof(args[0]),
+                                      (char **)args);
     ASSERT_EQ(app_config_->verbosity(), spdlog::level::level_enum::critical);
   }
   {
-    char const *args[] = {"/path/", "--verbosity", "6"};
-    app_config_->initialize_from_args(
-        AppConfiguration::LoadScheme::kFullSyncing,
-        sizeof(args) / sizeof(args[0]),
-        (char **)args);
+    char const *args[] = {"/path/",
+                          "--verbosity",
+                          "6",
```

Comment:
Perhaps needs test for various corner cases, including negative cases.
We can test bad value of **verbosity** here.